alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/LicenseMetrics.java

```diff
@@ -0,0 +1,118 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.service.descriptor.Descriptor;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.service.license.LicenseDescriptor;
+import org.alfresco.service.license.LicenseService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import javax.annotation.Nonnull;
+
+public class LicenseMetrics implements MeterBinder, ApplicationContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(LicenseMetrics.class);
+
+    private static final String METRIC_NAME_LICENSE = "license";
+
+    private ApplicationContext ctx;
+    private DescriptorService descriptorService;
+
+    public LicenseMetrics(DescriptorService descriptorService) {
+        this.descriptorService = descriptorService;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry registry) {
+        // do not do anything for Community
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        if(!"Enterprise".equals(serverDescriptor.getEdition())) {
+            logger.info("Edition={}, license metrics are not available", serverDescriptor.getEdition());
+            return;
+        }
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".valid", ctx, LicenseMetrics::getValid)
+                .description("Whether the license is still valid")
+                .register(registry);
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".remainingDays", descriptorService, LicenseMetrics::getRemainingDays)
+                .description("Remaining days")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxDocs", descriptorService, LicenseMetrics::getMaxDocs)
+                .description("Max docs")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxUsers", descriptorService, LicenseMetrics::getMaxUsers)
+                .description("Max users")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".isClusterEnabled", descriptorService, LicenseMetrics::isClusterEnabled)
```

Comment:
I'm not sure if it's useful to export these 3 metrics, they are more "constants" than metrics.

But I guess it doesn't really hurt either and if they take too much storage, we can still drop them during ingestion in prometheus.

Reply:
This is true, they change only with a change of license.

Reply:
Keep them please